staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/types_jsonschema.go

```diff
@@ -208,6 +210,20 @@ type ValidationRule struct {
 	// "x must be less than max ("+string(self.max)+")"
 	// +optional
 	MessageExpression string
+	// reason represents a machine-readable description of why this validation failed.
+	// If this is the first validation in the list to fail, this reason, as well as the
+	// corresponding HTTP response code, are used in the
+	// HTTP response to the client.
+	// The currently supported reasons are: "NotFound", "Forbidden", "Invalid", "RequestEntityTooLarge".
+	// If not set, "Invalid" is used in the response to the client.
+	// All status reason must be accepted by clients when reading this value.
+	// +optional
+	Reason *metav1.StatusReason
+	// jsonPath represents the field path returned when the validation failed.
+	// It is a relative JSON path (i.e. with array notation) scoped to the location of the x-kubernetes-validations extension in the schema.
+	// If not set, the current scoped field path will be returned back.
```

Comment:
I currently have it aligned with the JSONPath which is asking to start with a dot. Let me know if the other way is preferred. Will add an example.

Reply:
what do we return in the `field` path in field errors from normal validation? does the root path start with a dot?

Reply:
I'm not actually 100% sure what jsonPath specifies... if we call this jsonPath, we should match the spec if there is one

we should also do any transformations needed to make the returned field errors consistent with the ones we generate from declarative and built-in validation

Reply:
> I'm not actually 100% sure what jsonPath specifies...

TBH, me neither. But jsonPath is a big complex spec covering a lot more than we support. This is further complicated by the fact that CRDs already use jsonPath to declare fields (e.g. path to the scale fields, paths for table printer fields), _but_ the validation of those fields leaves a lot to be desired, as the documentation of what subset of jsonPath is allowed is also not great.

Since we're already using a subset of jsonPath in CRDs, I'm in favor of us clearly documenting what that subset is here. e.g. "jsonPath containing only... <description of what exact subset of jsonPath is allowed>" ?